### DISCUSSION TITLE
Best Practice for Command with On-the-Fly-Generated Paths?

### DISCUSSION THREAD

**Discussion by gcschmit** [mjansen4857/pathplanner#230] [Q&A] [answered]
If we are generating paths on the fly, is there a best practice regarding how to pass that path along with a command. The PPSwerveControllerCommand needs the path to be specified when constructed. We feel that we need a command that can be provided with the path after construction and prepares everything during initialization. We could implement our own command that would be almost identical to PPSwerveControllerCommand with this changed behavior but that feel clunky.

Suggestions?

**Comment by mjansen4857** [comment, marked as answer]
Pretty much everything in PPSwerveControllerCommand is prepared during initialization. The constructor doesn't do any actual work. Any memory being used (pretty negligible anyways) for each instance of the command will just be cleaned up after its done being used.

So, basically, you shouldn't have any problems with just constructing a new PPSwerveControllerCommand for each path you generate.

**Reply by gcschmit** [reply]
Sounds good. We will generate the path in response to a Trigger and, at that point, create a new PPSwerveControllerCommand and explicitly schedule it.
